luigi/worker.py

```diff
@@ -94,13 +94,14 @@ class TaskProcess(multiprocessing.Process):
     Mainly for convenience since this is run in a separate process. """
 
     def __init__(self, task, worker_id, result_queue, random_seed=False, worker_timeout=0,
-                 tracking_url_callback=None):
+                 tracking_url_callback=None, status_message_callback=None):
         super(TaskProcess, self).__init__()
         self.task = task
         self.worker_id = worker_id
         self.result_queue = result_queue
         self.random_seed = random_seed
         self.tracking_url_callback = tracking_url_callback
+        self.task._status_message_callback = status_message_callback
```

Comment:
Hmm. I wish we had done this trick for `tracking_url_callback` too...

Nice trick! :)